packages/in_app_purchase/in_app_purchase_storekit/lib/src/in_app_purchase_apis.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright 2013 The Flutter Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart' show visibleForTesting;
+
+import 'messages.g.dart';
+import 'sk2_pigeon.g.dart';
+
+/// The instance of the host API used to communicate with the platform side for
+/// the original StoreKit API.
+///
+/// This is a global to allow tests to override the host API with a mock, since
+/// in practice the host API is a singleton, and there is no way to inject it
+/// in the usual way since many uses aren't in objects.
+InAppPurchaseAPI hostApi = InAppPurchaseAPI();
```

Comment:
Can we mark the setter `@visibleForTesting` then?

Reply:
Ah it's already that way. But still I think these API instances are supposed to be read only outside of tests?